cpp/CMakeLists.txt

```diff
@@ -189,6 +190,13 @@ add_library(cudf SHARED
             src/groupby/groupby.cu
             src/groupby/new_groupby.cu
             src/binary/binary_ops.cu
+            src/binary/jit/code/kernel.cpp
```

Comment:
These additions _hard-code additional sources_ to the library, but I see that `JITIFY_PROCESS_CACHE` is an _option_. Should we make it so that these files are only added if the proper flag is set?

This would manifestly increase our build times.

Reply:
That's a misunderstanding. `JITIFY_PROCESS_CACHE` specifies that we should enable thread safety in Jitify and use a process level cache instead of a thread level cache. The code included in this PR would be active regardless of the option.

Reply:
👍